packages/expo-updates/e2e/fixtures/project_files/maestro/updates-server/update.ts

```diff
@@ -0,0 +1,224 @@
+  return {
+    id: crypto.randomUUID(),
+    createdAt: date.toISOString(),
+    runtimeVersion: RUNTIME_VERSION,
+    launchAsset: {
+      hash,
+      key,
+      contentType: 'application/javascript',
+      url: urlForBundleFilename(bundleFilename),
+    },
+    assets,
+    metadata: {},
+    extra: {
+      expoConfig: appJson.expo,
+    },
+  };
+}
+
+/**
+ * Method used in the fingerprint test to get the fingerprint to serve the update.
+ */
+async function getUpdateManifestForBundleFilenameWithFingerprintRuntimeVersionAsync(
+  date: { toISOString: () => string },
+  hash: string,
+  key: string,
+  bundleFilename: string,
+  assets: any[],
+  projectRoot: string,
+  platform: 'ios' | 'android'
+) {
+  const runtimeVersion = await getResolvedRuntimeVersionAsync(projectRoot, platform);
+  const appJson = require(`${projectRoot}/app.json`);
+  return {
+    id: crypto.randomUUID(),
+    createdAt: date.toISOString(),
+    runtimeVersion,
+    launchAsset: {
+      hash,
+      key,
+      contentType: 'application/javascript',
+      url: urlForBundleFilename(bundleFilename),
+    },
+    assets,
+    metadata: {},
+    extra: {
+      expoConfig: appJson.expo,
+    },
+  };
+}
+
+/**
+ * Common method used in all the tests to create valid rollback directives
+ */
+function getRollbackDirective(date: Date) {
+  return {
+    type: 'rollBackToEmbedded',
+    parameters: {
+      commitTime: date.toISOString(),
+    },
+  };
+}
+
+/**
+ * Common method used to create "no update available" directives
+ */
+function getNoUpdateAvailableDirective() {
+  return {
+    type: 'noUpdateAvailable',
+  };
+}
+
+async function getResolvedRuntimeVersionAsync(projectDir: string, platform: 'ios' | 'android') {
+  // change to true to get more detailed github output
+  const printDebug = false;
+
+  const resolvedRuntimeVersionJSONResult = await expoUpdatesCommandAsync(projectDir, [
+    'runtimeversion:resolve',
+    '--platform',
+    platform,
+    '--workflow',
+    'generic',
+    ...(printDebug ? ['--debug'] : []),
+  ]);
+  const runtimeVersionResult = JSON.parse(resolvedRuntimeVersionJSONResult);
+  if (printDebug) {
+    console.log('Resolved runtime version', resolvedRuntimeVersionJSONResult);
+  }
+  return nullthrows(runtimeVersionResult.runtimeVersion);
+}
+
+async function expoUpdatesCommandAsync(projectDir: string, args: string[]): Promise<string> {
+  const expoUpdatesCli =
+    silentResolveFrom(projectDir, 'expo-updates/bin/cli') ??
+    resolveFrom(projectDir, 'expo-updates/bin/cli.js');
+  return (await spawnAsync(expoUpdatesCli, args, { stdio: 'pipe' })).stdout;
+}
+
+const Update = {
+  copyBundleToStaticFolder,
+  copyAssetToStaticFolder,
+  findAssets,
+  getUpdateManifestForBundleFilename,
+  getUpdateManifestForBundleFilenameWithFingerprintRuntimeVersionAsync,
+  getRollbackDirective,
+  getNoUpdateAvailableDirective,
+  serverHost,
+  serverPort,
+};
+
+module.exports = { Update };
```

Comment:
can we just export each function than the module.exports here?

Reply:
I kept this pattern as it is the one used in the original files from the Detox tests.